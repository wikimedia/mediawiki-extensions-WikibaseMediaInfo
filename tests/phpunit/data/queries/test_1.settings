{
	"title": "Entities found, no lang fallbacks",
	"term": "some search term",
	"entitiesForTerm": {
		"some search term": [
			{
				"id": "Q1"
			},
			{
				"id": "Q2"
			}
		]
	},
	"userLanguage": "en",
	"settings": {
		"boost": {
			"statement": {"P1": 20, "P2": 20},
			"descriptions.$language": 7.5,
			"descriptions.$language.plain": 2.5,
			"category": 0.525,
			"category.plain": 0.175,
			"title": 0.45,
			"title.plain": 0.15,
			"redirect.title": 0.375,
			"redirect.title.plain": 0.125,
			"text": 0.3,
			"text.plain": 0.1,
			"auxiliary_text": 0.225,
			"auxiliary_text.plain": 0.075,
			"file_text": 0.15,
			"file_text.plain": 0.05
		}
	}
}
