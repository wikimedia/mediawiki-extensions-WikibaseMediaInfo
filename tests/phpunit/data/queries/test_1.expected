{
    "bool": {
        "must": [
            {
                "function_score": {
                    "query": {
                        "bool": {
                            "minimum_should_match": 1,
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 1,
                                        "queries": [
                                            {
                                                "bool": {
                                                    "minimum_should_match": 0,
                                                    "filter": [
                                                        {
                                                            "multi_match": {
                                                                "query": "some search term",
                                                                "fields": [
                                                                    "all",
                                                                    "all.plain"
                                                                ],
                                                                "operator": "and"
                                                            }
                                                        }
                                                    ],
                                                    "should": [
                                                        {
                                                            "match": {
                                                                "title": {
                                                                    "query": "some search term",
                                                                    "boost": 0.45
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "title.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.15
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "category": {
                                                                    "query": "some search term",
                                                                    "boost": 0.525
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "category.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.175
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "redirect.title": {
                                                                    "query": "some search term",
                                                                    "boost": 0.375
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "redirect.title.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.125
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "heading": {
                                                                    "query": "some search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "heading.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "auxiliary_text": {
                                                                    "query": "some search term",
                                                                    "boost": 0.225
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "auxiliary_text.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.075
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "text": {
                                                                    "query": "some search term",
                                                                    "boost": 0.3
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "text.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "file_text": {
                                                                    "query": "some search term",
                                                                    "boost": 0.15
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "file_text.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 0.05
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "suggest": {
                                                                    "query": "some search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "descriptions.en.plain": {
                                                                    "query": "some search term",
                                                                    "boost": 2.5
                                                                }
                                                            }
                                                        }
                                                    ]
                                                }
                                            }
                                        ]
                                    }
                                },
                                {
                                    "dis_max": {
                                        "queries": [
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P1=Q1",
                                                        "boost": 20
                                                    }
                                                }
                                            },
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P2=Q1",
                                                        "boost": 20
                                                    }
                                                }
                                            },
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P1=Q2",
                                                        "boost": 10
                                                    }
                                                }
                                            },
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P2=Q2",
                                                        "boost": 10
                                                    }
                                                }
                                            }
                                        ]
                                    }
                                }
                            ]
                        }
                    },
                    "functions": [
                        {
                            "script_score": {
                                "script": {
                                    "source": "100 \/ ( 1 + exp( -1 * ( _score + intercept ) ) )",
                                    "params": {
                                        "intercept": 1
                                    },
                                    "lang": "expression"
                                }
                            }
                        }
                    ]
                }
            }
        ],
        "filter": [
            {
                "terms": {
                    "namespace": [
                        0
                    ]
                }
            }
        ]
    }
}